src/main/java/net/wasdev/wlp/common/plugins/util/DevUtil.java

```diff
@@ -0,0 +1,956 @@
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package net.wasdev.wlp.common.plugins.util;
+
+import java.io.BufferedReader;
+import java.io.BufferedWriter;
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.io.FileReader;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.net.MalformedURLException;
+import java.net.URL;
+import java.net.URLClassLoader;
+import java.nio.charset.StandardCharsets;
+import java.nio.file.FileSystems;
+import java.nio.file.FileVisitResult;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.SimpleFileVisitor;
+import java.nio.file.StandardCopyOption;
+import java.nio.file.StandardWatchEventKinds;
+import java.nio.file.WatchEvent;
+import java.nio.file.WatchKey;
+import java.nio.file.WatchService;
+import java.nio.file.Watchable;
+import java.nio.file.attribute.BasicFileAttributes;
+import java.util.ArrayDeque;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.Deque;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Scanner;
+import java.util.Set;
+import java.util.concurrent.RejectedExecutionException;
+import java.util.concurrent.ThreadPoolExecutor;
+import java.util.concurrent.TimeUnit;
+import java.util.jar.Attributes;
+import java.util.jar.JarFile;
+import java.util.jar.Manifest;
+
+import javax.tools.JavaCompiler;
+import javax.tools.JavaFileObject;
+import javax.tools.StandardJavaFileManager;
+import javax.tools.StandardLocation;
+import javax.tools.ToolProvider;
+
+import org.apache.commons.io.FileUtils;
+
+import com.sun.nio.file.SensitivityWatchEventModifier;
+
+/**
+ * Utility class for dev mode.
+ */
+public abstract class DevUtil {
+
+    /**
+     * Log debug
+     * 
+     * @param msg
+     */
+    public abstract void debug(String msg);
+
+    /**
+     * Log debug
+     * 
+     * @param msg
+     * @param e
+     */
+    public abstract void debug(String msg, Throwable e);
+
+    /**
+     * Log debug
+     * 
+     * @param e
+     */
+    public abstract void debug(Throwable e);
+
+    /**
+     * Log warning
+     * 
+     * @param msg
+     */
+    public abstract void warn(String msg);
+
+    /**
+     * Log info
+     * 
+     * @param msg
+     */
+    public abstract void info(String msg);
+
+    /**
+     * Log error
+     * 
+     * @param msg
+     */
+    public abstract void error(String msg);
+
+    /**
+     * Returns whether debug is enabled by the current logger
+     * 
+     * @return whether debug is enabled
+     */
+    public abstract boolean isDebugEnabled();
+
+    /**
+     * Stop the server
+     */
+    public abstract void stopServer();
```

Comment:
Can `stopServer` and `startServer` be implemented using Ant directly at the common level? Based on the implementation in ci.maven, it seems like there are only a few parameters you need to retrieve from the build tool configuration. Those could simply be passed in to these methods as needed. The majority of the implementations for these methods in Maven seem like they would apply to Gradle as well.

Reply:
This does not seem specific to Dev Mode as other mojos (StartServer, StopServer, etc.) execute Ant tasks in ci.maven. If we were to implement `stopServer` and `startServer` for Dev Mode in ci.common it would make sense to make this change for every `stopServer` and `startServer` in ci.maven.  If appropriate we can open an issue to track making this change for all of ci.maven.

Reply:
The reason I suggest this is that the start server implementation for dev mode is much more involved than the existing implementations in `StartServerMojo` and `StopServerMojo` - particularly in setting up the WatchService for the logs. We can look into refactoring those into common, but for the most part they are just wrapping the ant task and performing little additional behavior.

Reply:
This would involve refactoring most of `StartDebugMojoSupport` into ci.common, which I think should be handled in a separate issue.

Reply:
Opened https://github.com/WASdev/ci.common/issues/54

Reply:
I did not intend to suggest refactoring what is in `StartDebugMojoSupport`. Rather, I would like to see what is in that method implementation in `DevMojo` in Maven moved to common. So initialize the `ServerTask` in common, and pass in whatever parameters that are needed from Maven/Gradle (`serverName`, `outputDir`) into `startServer()` to perform the initialization. Aside from the task initialization, I think the rest of what is in that method in Maven would apply to Gradle as well, right?

Reply:
Thanks, I see what you mean.  I've refactored `startServer` into ci.common in these commits: 
https://github.com/WASdev/ci.maven/pull/476/commits/7ec10c9256d90943e48b13e898965bd1b9252e63 
https://github.com/WASdev/ci.common/pull/48/commits/e5747ef4b3f812dd3640c35897642bd7c88bbd08 
https://github.com/WASdev/ci.common/pull/48/commits/b8149f14aca5d3eb20a251a18712a622fb79e897

Didn't refactor `stopServer` since that's just a few lines and would have no benefit because it still needs to call `initializeJava` in ci.maven.